cyder/base/fields.py

```diff
@@ -51,12 +51,15 @@ def to_python(self, value):
         value = super(MacAddrField, self).to_python(value)
 
         if value:
-            value = value.lower().replace(':', '').replace('-', '')
-            value = reduce(lambda x,y: x + ':' + y,
-                           (value[i:i+2] for i in xrange(0, 12, 2)))
-        elif value == '':
-            value = None
-        return value
+            raw = value.lower().replace(':', '').replace('-', '')
+            if len(raw) != 12:
+                # We can't raise a ValidationError in to_python() (because
+                # nothing will catch it), so all we can do is make sure clean()
+                # will raise a ValidationError when it's called next.
+                return 'TOO LONG'
```

Comment:
If I understand correctly, the value returned by this method will never be seen by users... it's just a dummy value that indicates to anyone reading the code that it is invalid.